dspace-server-webapp/src/test/java/org/dspace/app/rest/RestResourceControllerIT.java

```diff
@@ -71,4 +73,24 @@ public void notExistentResourceValidSubPath() throws Exception {
                 .andExpect(status().isNotFound());
     }
 
+    @Test
+    public void selfLinkContainsRequestParametersWhenProvided() throws Exception {
+        // When we call the root endpoint
+        getClient().perform(get("/api/core/metadatafields/search/byFieldName?schema=dc&offset=0"))
+                   // The status has to be 404 Not Found
+                   .andExpect(jsonPath("$._links.self.href", endsWith(
+                           "/api/core/metadatafields/search/byFieldName?schema=dc&offset=0")));
+    }
+
+    @Test
+    public void selfLinkDevoidOfRequestParametersWhenNoneProvided() throws Exception {
+        // When we call the root endpoint
+        getClient().perform(get("/api/core/metadatafields/search/byFieldName"))
+                   // The status has to be 404 Not Found
+                   .andExpect(jsonPath("$._links.self.href",
+                                       endsWith("/api/core/metadatafields/search/byFieldName")));
```

Comment:
please update / remote the comments as they don't apply to these tests

Reply:
@abollini Good catch, I'd copied the previous tests, but neglected to update the code docs.
Update these accordingly